tests/units/data/test_torch_dataset.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+
+import torch
+
+from docarray import BaseDocument, DocumentArray
+from docarray.data import TorchDataset
+from docarray.documents import Image, Text
+
+
+class ImagePreprocess:
+    def __call__(self, image: Image) -> None:
+        assert isinstance(image, Image)
+        image.tensor = torch.randn(3, 32, 32)
+
+
+class TextPreprocess:
+    def __call__(self, text: Text) -> None:
+        assert isinstance(text, Text)
+        text.embedding = torch.randn(64)
+
+
+def test_torch_dataset():
+    BATCH_SIZE = 32
+
+    class PairTextImage(BaseDocument):
+        text: Text
+        image: Image
+
+    class Mydataset(TorchDataset[PairTextImage]):
+        def __init__(self, csv_file: str, preprocessing: Optional[dict] = None):
+            with open(csv_file, "r") as f:
+                f.readline()
+                da = DocumentArray[PairTextImage](
+                    PairTextImage(
+                        text=Text(text=i[1]),
+                        image=Image(url=f"tests/toydata/image-data/{i[0]}"),
+                    )
+                    for i in map(lambda x: x.strip().split(","), f.readlines())
+                )
+            super().__init__(da, preprocessing)
```

Comment:
can we switch this test to directly use `TorchDataset[PairTextImage]` se we reflect this pattern that we want to promote?
(actually this would probably also not pass mypy now that I think about it, since the init args are different)

Reply:
Yup. Done.
Does it not pass mypy? I tried it and it seems fine.

Reply:
you ran mypy on this test file? Then I'm wrong haha, I expected it to fail